platform/android/src/geojson/feature_collection.cpp

```diff
@@ -0,0 +1,40 @@
+#include "feature_collection.hpp"
+
+#include "feature.hpp"
+
+namespace mbgl {
+namespace android {
+namespace geojson {
+
+mbgl::FeatureCollection FeatureCollection::convert(jni::JNIEnv& env, jni::Object<FeatureCollection> jCollection) {
+    auto jFeatureList = FeatureCollection::getFeatures(env, jCollection);
+    auto jFeatures = java::util::List::toArray<Feature>(env, jFeatureList);
```

Comment:
Have we ever benchmarked:

1. Using `List::toArray()` and then iterating over the result, extracting individual elements via `GetArrayRegion`?
2. Using `List::toArray()`, then converting the entire array to `std::vector` (or raw array) via `GetArrayRegion`, then iterating over that?
3. Using `List::toArray()`, iterating over reasonably-sized chunks copied to `std::vector` (or raw array) via `GetArrayRegion`?
4. Iterating over the `List` directly via JNI access to `List::iterator`?

[This article](https://www.ibm.com/developerworks/library/j-jni/) suggests that (3) is the best approach. We could package that up as a reusable template in jni.hpp that handles both chunking and deleting the temporary local refs.